sdks/python/apache_beam/runners/interactive/sql/beam_sql_magics.py

```diff
@@ -110,15 +126,15 @@ def beam_sql(self, line: str, cell: str) -> Union[None, PValue]:
         return
       register_coder_for_schema(pcoll.element_type)
 
-    # TODO(BEAM-10708): implicitly execute the pipeline and write output into
-    # cache.
-    return apply_sql(cell, line, found)
+    output_name, output = apply_sql(cell, line, found)
+    cache_output(output_name, output)
```

Comment:
I can imagine the use case that a user might be modifying the DB in a separate application and they want the most recent view instead of a cached view. How might this use case be supported with this caching logic?

Reply:
All the magics are one-shots. The user could re-execute the cell with the magic again to take in the most recent view.
The cache here serves as the medium to materialize the output PCollection's ​data for introspection.

We should notify the user about the diverge of pipelines after the magic in notebook examples we publish since the output PCollection is no longer part of their user pipeline.  The ideal usage of the magic is collecting as many sources as possible,  writing a pipeline in SQL using all the sources and introspecting the query's output, then if needed, sink the output PCollection to somewhere.

Reply:
ack